hassio/hassio-main.html

```diff
@@ -44,58 +51,77 @@
       ></hass-loading-screen>
     </template>
     <template is='dom-if' if='[[loaded]]'>
-      <iron-pages
-        selected='[[_routeData.page]]'
-        attr-for-selected='page-name'
-        fallback-selection='not-found'
-        selected-attribute='visible'
-      >
-        <hassio-addon-view
-          page-name='addon'
-          route='[[route]]'
-          hass='[[hass]]'
-        ></hassio-addon-view>
-
-        <hassio-addon-store
-          page-name='store'
-          hass='[[hass]]'
-        ></hassio-addon-store>
-
-        <hassio-supervisor
-          page-name='supervisor'
-          hass='[[hass]]'
-          supervisor-info='[[supervisorInfo]]'
-        ></hassio-supervisor>
-
-        <hassio-dashboard
-          page-name='dashboard'
-          hass='[[hass]]'
-          narrow='[[narrow]]'
-          show-menu='[[showMenu]]'
-          supervisor-info='[[supervisorInfo]]'
-          hass-info='[[hassInfo]]'
-        ></hassio-dashboard>
-
-        <hassio-advanced
-          page-name='advanced'
-          hass='[[hass]]'
-          supervisor-info='[[supervisorInfo]]'
-          host-info='[[hostInfo]]'
-          hass-info='[[hassInfo]]'
-        ></hassio-advanced>
-
-        <hassio-snapshots
-          page-name='snapshot'
-          hass='[[hass]]'
-          installed-addons='[[supervisorInfo.addons]]'
-        ></hassio-snapshots>
-
-        <hass-error-screen
-          page-name='not-found'
-          error='Page not found.'
-          title='Hass.io'
-        ></hass-error-screen>
-      </iron-pages>
+      <app-header-layout has-scrolling-region>
+        <app-header fixed slot='header'>
+          <app-toolbar>            
+            <ha-menu-button narrow='[[narrow]]' show-menu='[[showMenu]]'></ha-menu-button>
+            <paper-icon-button
```

Comment:
I think that it is bad design to have 1 component do two things: how can something both be a "root" nav tab component and also be a component that shows the details of something? Better split it out in 3 components:

navigator
 - show page with tabs
 - show details of X